### DISCUSSION TITLE
Beginner SWE, Grew up on J&D, would LOVE to help.

### DISCUSSION THREAD

**Discussion by JacobDChamberlain** [open-goal/jak-project#3538] [Q&A]
Hello friends,

I've been a software developer for almost 2 years professionally, and my job is boring af and I'd love to contribute to an open-source project I find meaningful in my downtime. I LOVE the Jak & Daxter games; they were a huge part of my upbringing. I don't know where to begin helping out. I have experience with general backend dev and some frontend, no game programming, but I'm a quick learner. 

Will someone please point me in the right direction of where I can be useful?

Thank you so much!!

**Comment by xTVaser** [comment]
Hey there, this question has been asked and discussed a few times in our Discord so you may find more information there (or not) but it's a tricky question to answer as we don't really have a list of "easy first issues" and realistically you would want to latch onto something that you are actually interested in working on.

So I'll just list a bunch of things that come to mind and if something sounds interesting to you, I would recommend taking a stab at it and if you get stuck, feel free to ask questions in discord and if people have time to spare they may assist.

In no particular order:
- Check out the issues pages for our repos and see if something feels approachable/interesting
  - https://github.com/open-goal/jak-project/issues is largely focused around the games, and the programming language.  Therefore it's OpenGOAL code (the games) and C++ code (the compiler and decompiler).
  - https://github.com/open-goal/launcher the launcher that helps people install the game is more approachable, using Rust + Web libraries to build a simple frontend
- Right now Jak 3 has basically been fully decompiled, but there are many bugs that need to be squashed.
  - If you are interested in learning and writing OpenGOAL, this could be a good starting point (depending on how tricky the bug is) we try to maintain a list of bugs here https://github.com/orgs/open-goal/projects/3/views/13
  - Alternatively, if you want to find bugs, that's also an idea.
- You may find it exciting to try your hand at making mods for the games, there are several people that are patient and willing to help as time permits, and they've put together very detailed example docs around it https://opengoal.dev/docs/category/modding-examples
- Documentation improvements aren't as glamorous but are appreciated https://github.com/open-goal/open-goal.github.io

All of that is to say that if you do decide to tackle something, just make sure you are truly interested/invested in it, as anything that involves working on the games will have a steep learning curve at first.  But there is no shortage of work, though typically the easier work items aren't as attractive 😄

**Reply by JacobDChamberlain** [reply]
Thank you so much!
